node/send_receive/lib/client.js

```diff
@@ -16,11 +16,14 @@ var MessagingEntityNotFoundError = require('./errors.js').MessagingEntityNotFoun
  * Instantiate a client pointing to the Event Hub given by this configuration.
  *
  * @param {ConnectionConfig} config
+ * @param {object} [policyOverrides] Optional overrides for node-amqp10 policy fields, see https://github.com/noodlefrenzy/node-amqp10 for details.
  * @constructor
  */
-function EventHubClient(config) {
+function EventHubClient(config, policyOverrides) {
```

Comment:
I completely subscribe to the idea of protocol-specific options (as in AMQP, HTTP, MQTT). it's something we need in IoT Hub too. I'm a little more cautious about transport-specific options (as in "transport library" : node-amqp10 in this case).

The reason is, it makes the code instantiating `EventHubClient` code tightly coupled with the `amqp10` underlying dependency, meaning if you guys change something later (say, policy property names), or if we add more transports, user code could break even though the user is unaware of the amqp10 underpinnings, thinking he/she's just playing with event hubs sdk options (but maybe I worry too much, the options are clearly flagged node-amqp10... I might be talking philosophy more than facts here).

Another route to consider that might get us to the same result is use plain and simple IoC. Since the policy object is a constructor parameter of `amqp10.Client`, we could instead have the optional argument be the `amqp10.Client` itself: 
- either the event hubs sdk user is unaware of the transport-specific features and doesn't pass the argument: we create a "default transport object" with "default policies".
- or the user wants to use specific features of the transport, then he can inject his preconfigured `amqp10.Client` instance.
That way we appear transport-agnostic for basic scenarios, and the user consciously takes a dependency on amqp10 when using advanced features.

The other thing that bugs me (might be my ignorance speaking though) is that by surfacing transport features we're creating feature imparity between SDKs (node, java, c#...), so it requires vetting this with a PM. Whereas if we inject the transport object, it's obvious that this is a transport feature and not and Event Hubs feature therefore we don't have to ask anyone or worry about SDK feature parity :)